proposals/v3/api.md

```diff
@@ -0,0 +1,1119 @@
+# Synse v3 HTTP API
+## Summary
+The driving change for Synse v3 is removing `rack/board/device`-based routing
+and using a new [tag](tags.md)-based routing system. This change is reflected in
+the API by replacing the old routing components with the new ones. In addition,
+various API improvements are included. These improvements are both a result of the
+new routing system and a result of feedback from v2 API usage.
+
+## High Level Work Items
+- Update the HTTP API in accordance with the API spec detailed in this document
+- Look into using [swagger](https://swagger.io/) for API documentation generation
+
+## Proposal
+This section contains the proposed API specification for Synse v3.
+
+Following the pattern established in previous versions of Synse Server,
+the URI prefix for requests will be `/synse/{API_VERSION}`, except for the two
+unversioned endpoints (`/test` and `/version`) where the URI prefix is just
+`/synse`.
+
+### Errors
+Most errors returned from Synse Server will come back with a JSON payload in order to
+provide additional context for the error. Some errors will not return a JSON payload;
+this class of error is generally due to the service not being ready, available, or
+reachable.
+
+#### HTTP Codes
+An error response will be returned with one of the following HTTP codes:
+
+* **400**: Invalid user input. This can range from invalid POSTed JSON, unsupported query
+  parameters being used, or invalid resource types.
+* **404**: Invalid URL or the specified resource is not found.
+* **500**: Server-side processing error.
+
+#### Scheme
+Below is an example of an error's JSON response:
+
+```json
+{
+  "http_code": 404,
+  "error_id": 4000,
```

Comment:
If the user does not need it, then let's not publicize it, otherwise the customer may rely on it.

Or - we completely break the interface at will and alienate our customers.

Reply:
removed `error_id` from error response scheme in e4a4cb6